test/functional/README.md

```diff
@@ -165,6 +165,23 @@ To include tests to be run in the CI system:
 TBD  
 <br/>
 
+### Debugging Tests
+
+As mentioned in the [test principles](#test-principles), automated tests should
+create their own test environment including all needed dependencies when the test
+starts, and should clean up after themselves once the test is over. This means
+the test environment will be deleted as soon as the test finishes running.
+Sometimes this is not the desired behavior, for example when writing a new test
+and is not working as expected, in this case it would be very difficult to figure
```

Comment:
This sentence needs re-wording. You don't specify what you are referring to when you say `for example when writing a new test and is not working as expected`. You have to specify what is not working as expected in that sentence: `for example when writing a new test and the test is not working as expected`. You can also drop the `in this case` if you leave it part of that same sentence, since the case was already described. Something like

> Sometimes this is not the desired behavior, for example when writing a new test and the test is not
> working as expected it would be very difficult to debug the problem if the test environment is wiped.

But if I wrote it I would say something like this

> Sometimes it is necessary to debug faulty tests, and therefore a persistent test environment is required.

Reply:
I like it, I'll make the change